src/handwritten/errors.py

```diff
@@ -45,18 +45,24 @@ def __init__(self, message, error_code, task_name=''):
             message (string): Specifies the error message.
             error_code (int): Specifies the NI-DAQmx error code.
         """
-        if task_name:
-            message = f'{message}\n\nTask Name: {task_name}'
-
-        super().__init__(message)
-
         self._error_code = int(error_code)
 
         try:
             self._error_type = DAQmxErrors(self._error_code)
         except ValueError:
             self._error_type = DAQmxErrors.UNKNOWN
 
+        # If message is empty, we try to put at least some information in it
+        if not message and self._error_type != DAQmxErrors.UNKNOWN:
+            message = f'{self._error_type.name} occurred. Error code: {self._error_code}.'
+        elif not message:
+            message = f'Error code: {self._error_code}.'
+
+        if task_name:
+            message = f'{message}\n\nTask Name: {task_name}'
```

Comment:
I think you should invert this to better match .NET

![image](https://github.com/ni/nidaqmx-python/assets/42776655/e7ba4981-83aa-4da3-ba66-b9528fbac3db)

pseudocode:

```python
message = f"{message}\n\nTask Name: {task_name}\n\nStatus: {self._error_type.name} ({self._error_code})"
```

Reply:
Changed